### PR TITLE
5th lesson (batch API)

### PR DESCRIPTION
**Pull request by dakruchko**

### CODE REVIEW

File: sites/all/modules/custom/lesson_batch/lesson_batch.batch.inc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Process every item in batch
+ */
+function lesson_batch_process_currencies($data, &$context) {
+  $data_to_process = json_decode($data, TRUE);
+  $limit = 1;
+  if (empty($context['sandbox']['progress'])) {
+    $context['sandbox']['progress'] = 0;
+    $context['sandbox']['max'] = count($data_to_process);
+  }
+  if(empty($context['sandbox']['items'])) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_batch/lesson_batch.batch.inc]: Code style

File: sites/all/modules/custom/lesson_batch/lesson_batch.module [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      ],
+    ];
+    drupal_add_js($modal_style, 'setting');
+  }
+}
+
+/**
+ * Sends request to fetch currency rates.
+ */
+function lesson_batch_get_currency_rates() {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_batch/lesson_batch.module:101]: Handle the case when the endpoint will be unavailable (request returns error code)

File: sites/all/modules/custom/lesson_batch/lesson_batch.pages.inc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ */
+
+/**
+ * Page callback: Displays a a table with uploaded files.
+ *
+ * @see lesson_batch_menu().
+ */
+function lesson_batch_landing_table() {
+  drupal_add_js(drupal_get_path('module', 'lesson_batch') . '/js/lesson_batch.js');
+  drupal_add_js(array('uid' => $GLOBALS['user']->uid, ), 'setting');
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_batch/lesson_batch.pages.inc]: The best way to work with global User object is
global $user;
$a = $user->uid;

File: sites/all/modules/custom/lesson_batch/lesson_batch.pages.inc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  $uploaded_files = [];
+  $uploaded_files = db_select('file_managed', 'fm')
+    ->fields('fm', ['fid', 'uid', 'uri', 'status'])
+    ->execute()
+    ->fetchAll();
+  $user_has_rights_to_download = FALSE;
+  foreach ($uploaded_files as $file) {
+    if ($file->uid === $GLOBALS['user']->uid) {
+      $user_has_rights_to_download = TRUE;
+      break;
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_batch/lesson_batch.pages.inc:25]: Code styles

Comment (dakruchko) [reply sites/all/modules/custom/lesson_batch/lesson_batch.pages.inc:25]: Sorry, you mean that variable name is a little weird?

**Review by dwebpoint — APPROVED** [review]

**Pull request merged**